bin/1-create-artifact-bucket.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+set -eo pipefail
+
+BUCKET_NAME=$1
+
+if [[ -z "$BUCKET_NAME" ]]; then
+  BUCKET_ID=$(dd if=/dev/random bs=8 count=1 2>/dev/null | od -An -tx1 | tr -d ' \t\n')
+  BUCKET_NAME=anlessini-lambda-artifacts-$BUCKET_ID
+  echo "BUCKET_NAME is not provided, fall back to randomly generated: $BUCKET_NAME"
+else
+  echo "BUCKET_NAME: $BUCKET_NAME"
+fi
+
+aws s3 mb s3://$BUCKET_NAME
```

Comment:
I think we can do this step in SAM (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket.html), and that way we can refer to it in the SAM in the environment variables given to the lambda. On an unrelated side note, we can also provision dynamoDB resources in SAM too.

Reply:
You might be able to do that for the index bucket, but I don't think you can provision artifact bucket for Lambda using cloudformation.

Reply:
I will be removing a lot of my resource provisioning related changes in this PR. I shouldn't have included them in the first place since they are not the focus of this PR.